src/inventory.gql

```diff
@@ -16,12 +16,44 @@ type ProductItem {
 """A simple UUID parser"""
 scalar UUID
 
+type PaginatedProductItem {
+  nodes: [ProductItem!]
+  totalCount: Int!
+  hasNextPage: Boolean!
+}
+
 type Query {
-  productItems: [ProductItem!]!
+  productItems(findProductItemsInput: FindProductItemArgs!): PaginatedProductItem!
```

Comment:
this should take the inputs directly and not as an input object

Reply:
Wait, why?
Isn't that what the GraphQL docs recommend?

Reply:
For mutations yes
However for these field inputs it is not common.
Don't ask me why